### PR TITLE
Proxy2

### PR DESCRIPTION
**Pull request by bretambrose**
* Proxyv2 bindings
* Proxy integration tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awscrt/http.py [hunk 1]
```diff
@@ -532,6 +532,20 @@ def __str__(self):
         return self.__class__.__name__ + "(" + str([pair for pair in self]) + ")"
 
 
+class HttpProxyConnectionType(IntEnum):
+    """Proxy connection type enumeration"""
+    Legacy = 0
+    """Use the old connection establishment logic that would use
+         (1) Forwarding if not using tls
+         (2) Tunneling if using tls"""
```
Comment (graebm) [inline awscrt/http.py:540]: Update the HTML docs by running `./make-docs.sh`
Look at the result by opening `docs/index.html` to see if the formatting looks good

you might need to do like:
```
1. Forward ifblahbalh
2. Tunneling if blahlbalh
```
check the [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks) if that doesn't work

File: awscrt/http.py [hunk 2]
```diff
@@ -532,6 +532,20 @@ def __str__(self):
...
+
+    Forwarding = 1
+    """Establish a request forwarding connection to the proxy.  In this case, tls is not a valid option."""
```
Comment (graebm) [inline awscrt/http.py:543]: nitpick: capitalize TLS in these comments

File: awscrt/mqtt.py [hunk 3]
```diff
@@ -228,7 +228,7 @@ class Connection(NativeResource):
         use_websocket (bool): If true, connect to MQTT over websockets.
 
         websocket_proxy_options (Optional[awscrt.http.HttpProxyOptions]):
-            Optional proxy options for websocket connections.
+            Optional proxy options for websocket connections.  Deprecated by proxy_options.
```
Comment (graebm) [inline awscrt/mqtt.py:231]: nitpick: How about just
```
Deprecated, use `proxy_options` instead.
```

File: test/test_proxy.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    def get_port_from_test_type(test_type):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return 80
+        else:
+            return 443
+
+
+class ProxyHttpTest(NativeResourceTest):
+
+    timeout = 10  # seconds
```
Comment (graebm) [inline test/test_proxy.py:163]: just use our default `TIMEOUT` you already imported at the top of the file:
`from test import NativeResourceTest, TIMEOUT`

File: test/test_proxy.py [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(200, stream_completion_result)
+
+        return
+
+    @unittest.skipIf(not ProxyTestConfiguration.is_proxy_environment_initialized(), 'requires proxy test env vars')
+    def test_forwarding_proxy_no_auth(self):
+        self._do_proxy_http_test(ProxyTestType.FORWARDING, HttpProxyAuthenticationType.Nothing)
+        return
```
Comment (graebm) [inline test/test_proxy.py:202]: nitpick: delete all these blank `return` statements. they're not needed, and they look weird

File: awscrt/mqtt.py [hunk 6]
```diff
@@ -241,6 +241,9 @@ class Connection(NativeResource):
                     `transform_args.done()` when complete.
 
                 *   `**kwargs` (dict): Forward-compatibility kwargs.
+
+        proxy_options (Optional[awscrt.http.HttpProxyOptions]):
+            Optional proxy options for all connections.  Supercedes websocket_proxy_options.
```
Comment (graebm) [inline awscrt/mqtt.py:246]: nitpick: why mention websocket_proxy_options at all? It's dead to us now

**Review by graebm — APPROVED** [review]
fix & ship

File: awscrt/http.py (hunk 1 above)
Comment (bretambrose) [reply awscrt/http.py:540]: I'm getting the following:

reading sources... [100%] index                                                                                                            

Sphinx error:
master file /home/bret/projects/crt/aws-crt-python/docsrc/source/contents.rst not found

**Pull request merged**